src/calendar/tests/CalendarDate.spec.js

```diff
@@ -0,0 +1,463 @@
+import React from 'react/addons';
+import moment from 'moment';
+import _ from 'underscore';
+
+import CalendarDate from '../CalendarDate';
+
+import CalendarDatePeriod from '../CalendarDatePeriod';
+import CalendarHighlight from '../CalendarHighlight';
+import CalendarSelection from '../CalendarSelection';
+
+const TestUtils = React.addons.TestUtils;
+
+describe('The CalendarDate Component', function () {
+
+  beforeEach(function () {
+
+    const getCalendarDate = (props) => {
+
+      props = _.extend({
+        date: moment(),
+        firstOfMonth: moment(),
+        dateRangesForDate: function () {
+          return {
+            count: function () {
+              return props.count || 1;
+            },
+            getIn: function (data) {
+              if (data[0] === 0) {
```

Comment:
:question: What are these tests - what are we testing here?

Could we have a named predicate for this to make it easier to the next reader what this should do ? Without additional context it's a bit unclear currently

```
const isSomething = (data[0] === 0);
if (isSomething) { 
   ....
```